com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/activiti/RetryManagementProcessAction.java

```diff
@@ -0,0 +1,44 @@
+package com.sap.cloud.lm.sl.cf.core.activiti;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.flowable.engine.runtime.Execution;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sap.cloud.lm.sl.cf.persistence.services.ProgressMessageService;
+import com.sap.cloud.lm.sl.common.SLException;
+
+public class RetryManagementProcessAction extends ProcessAction {
```

Comment:
Remove the hierarchy and replace it with a list of additional actions, which is given to each ProcessAction.